### PR TITLE
V8 Migration!

### PR DESCRIPTION
**Pull request by JayVora-SerpentCS**

### CODE REVIEW

File: tests/test_connector_redmine.py [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+class TestRedmineConnector(TransactionCase):
+    def setUp(self):
+        super(TestRedmineConnector, self).setUp()
+        self.backend_model = self.registry('redmine.backend')
+        self.user_model = self.registry("res.users")
+        self.employee_model = self.registry('hr.employee')
+        self.timesheet_model = self.registry('hr.analytic.timesheet')
+        self.account_model = self.registry('account.analytic.account')
+        self.redmine_model = self.registry('redmine.hr.analytic.timesheet')
+
```
Comment (ghost) [inline tests/test_connector_redmine.py:43]: Please use the new api in this file.
self.env['...']
remove cr, uid, context

File: tests/test_connector_redmine.py [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+        self.user_id = self.user_model.create(
+            cr, uid, {
+                'name': 'User 1',
+                'login': 'user_1',
+            }, context=context)
+
+        self.user = self.user_model.browse(
+            cr, uid, self.user_id, context=context)
+
+        journal_id = self.registry('ir.model.data').get_object_reference(
```
Comment (ghost) [inline tests/test_connector_redmine.py]: journal = self.env.ref('...')

File: models/redmine_hr_analytic_timesheet.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+from openerp import models, fields
+
+class RedmineTimeEntry(models.Model):
+    _name = 'redmine.hr.analytic.timesheet'
+    _description = 'Redmine Time Entry Binding'
+    _inherit = 'redmine.binding'
+#    _inherits = {'hr.analytic.timesheet': 'openerp_id'}
+
+    openerp_id = fields.Many2one('hr.analytic.timesheet', 'Timesheet', delegate=True, required=True, ondelete='cascade')
```
Comment (ghost) [inline models/redmine_hr_analytic_timesheet.py]: Watch out for PEP8. max 80 char per line

File: models/redmine_hr_analytic_timesheet.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+#
+##############################################################################
+
+from openerp import models, fields
+
+class RedmineTimeEntry(models.Model):
+    _name = 'redmine.hr.analytic.timesheet'
+    _description = 'Redmine Time Entry Binding'
+    _inherit = 'redmine.binding'
+#    _inherits = {'hr.analytic.timesheet': 'openerp_id'}
```
Comment (ghost) [inline models/redmine_hr_analytic_timesheet.py]: If this is not required anymore, please remove it.

File: tests/test_connector_redmine.py [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp.tests.common import TransactionCase
+from openerp.tools import (DEFAULT_SERVER_DATETIME_FORMAT, DEFAULT_SERVER_DATE_FORMAT)
```
Comment (ghost) [inline tests/test_connector_redmine.py]: PEP8 here as well

File: unit/__init__.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+import backend_adapter
+import binder
+import import_synchronizer
+import mapper
```
Comment (ghost) [inline unit/__init__.py]: use relative import
from . import ...

File: unit/backend_adapter.py [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp.tools.translate import _
+from openerp.addons.connector.exception import (NetworkRetryableError, FailedJobError, InvalidDataError)
```
Comment (ghost) [inline unit/backend_adapter.py]: PEP8

File: unit/import_synchronizer.py [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+import logging
+from openerp.addons.connector.unit.synchronizer import ImportSynchronizer
+from openerp.addons.connector.queue.job import job
+from ..import connector
```
Comment (ghost) [inline unit/import_synchronizer.py:25]: missing white space after ..

File: unit/backend_adapter.py [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+                    'Invalid authentications key.'))
+
+        except (exceptions.UnknownError, exceptions.ServerError) as err:
+            raise NetworkRetryableError(
+                _('A network error caused the failure of the job: '
+                    '%s') % ustr(err))
+
+        self.redmine_api = redmine_api
+
+    def search_user(self,login):
```
Comment (ghost) [inline unit/backend_adapter.py]: missing white space, please use a python linther

File: models/redmine_binding.py [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+##############################################################################
+from openerp import models, fields
+
+class RedmineBinding(models.AbstractModel):
+
+    _name = 'redmine.binding'
+    _inherit = 'external.binding'
+    _description = 'Redmine Binding (Abstract)'
+
+    backend_id = fields.Many2one('redmine.backend', 'Redmine Backend', required=True, ondelete='restrict')
```
Comment (ghost) [inline models/redmine_binding.py]: PEP8

File: models/redmine_backend.py [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+                 "Note that a similar configuration exists "
+                 "for each storeview.")
+
+    @api.multi
+    def _get_base_adapter(self):
+        """
+        Get an adapter to test the backend connection
+        """
+        session = ConnectorSession(self._cr, self._uid, context=self._context)
+        environment = ConnectorEnvironment(self[0], session, None)
```
Comment (ghost) [inline models/redmine_backend.py]: remove [0]
use self.ensure_one()

File: models/redmine_backend.py [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+
+
+    location = fields.Char('Location', size=128, required=True)
+    key = fields.Char('Key', size=64, required=True)
+    version = fields.Selection('_select_versions', string='Version', required=True)
+    default_lang_id = fields.Many2one('res.lang', 'Default Language',
+            help="If a default language is selected, the records "
+                 "will be imported in the translation of this language.\n"
+                 "Note that a similar configuration exists "
+                 "for each storeview.")
```
Comment (ghost) [inline models/redmine_backend.py]: Bad indentation. PEP8 E128.

File: __openerp__.py [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+{
+    'name': 'Redmine Connector',
+    'version': '0.1',
+    'author': 'Savoir-faire Linux,Odoo Community Association (OCA),Serpent Consulting Services Pvt Ltd',
```
Comment (ghost) [inline __openerp__.py]: PEP8

File: __openerp__.py [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+    'maintainer': 'Odoo Community Association (OCA)',
+    'website': 'http://odoo-community.org',
+    'category': 'Connector',
+    'description': """
+Redmine Connector
+=================
+Add the Redmine Backend
+""",
```
Comment (ghost) [inline __openerp__.py]: description is not required on v8. You may remove it.

File: README.rst [hunk 15]
```diff
@@ -0,0 +1,64 @@
+Redmine Connector
```
Comment (yvaucher) [inline README.rst]: Please add AGPL logo on top

File: __init__.py [hunk 16]
```diff
@@ -0,0 +1,25 @@
...
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+import models
+import unit
+import backend
+import connector
```
Comment (yvaucher) [inline __init__.py]: Please respect POSIX standard putting a line break on last line of file

File: __openerp__.py [hunk 17]
```diff
@@ -0,0 +1,52 @@
...
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+{
+    'name': 'Redmine Connector',
+    'version': '0.1',
```
Comment (yvaucher) [inline __openerp__.py]: Please put full version number:

8.0.0.1 or 8.0.0.1.0

File: tests/test_connector_redmine.py [hunk 18]
```diff
@@ -0,0 +1,217 @@
...
+        }, context=context)
+
+        self.backend = self.backend_model.browse(
+            cr, uid, self.backend_id, context=context)
+
+        self.session = ConnectorSession(cr, uid, context=context)
+        self.environment = ConnectorEnvironment(
+            self.backend, self.session, 'redmine.hr.analytic.timesheet')
+
+        self.now = datetime.now().strftime(DEFAULT_SERVER_DATETIME_FORMAT)
```
Comment (yvaucher) [inline tests/test_connector_redmine.py:85]: For dates, use `fields.Datetime` and `fields.Date` helpers (`from_string()`, `now()`, ...)
https://www.odoo.com/documentation/8.0/reference/orm.html#openerp.fields.Date

File: views/redmine_backend_view.xml [hunk 19]
```diff
@@ -0,0 +1,55 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
+  <data>
+    <record id="view_redmine_backend_form" model="ir.ui.view">
+      <field name="name">redmine.backend.form</field>
+      <field name="model">redmine.backend</field>
+      <field name="arch" type="xml">
+        <form string="Redmine Backend" version="7.0">
```
Comment (yvaucher) [inline views/redmine_backend_view.xml]: You can remove `version="7.0"` from views

**Comment by yvaucher** [comment]
Few little things to fix to make pylint happy plus see my comments

**Comment by JayVora-SerpentCS** [comment]
Noted, we should take care of it ASAP.

**Comment by jtraulle** [comment]
@JayVora-SerpentCS What is the status of this merge request ?
I tried to install your current branch into a test environnement of Odoo 8.0 but, after adding Redmine backend, I do not understand what to do next. Maybe it is because it is not done ?

**Comment by JayVora-SerpentCS** [comment]
@jtraulle haven't been able to spend time to push the code. I will add the team to do this..

File: tests/__init__.py [hunk 20]
```diff
@@ -0,0 +1,25 @@
...
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from . import test_connector_redmine
+
+checks = [test_connector_redmine]
```
Comment (ghost) [inline tests/__init__.py:25]: you may remove checks

**Comment by ghost** [comment]
@JayVora-SerpentCS I figured that instead of a standard project with multiple modules, you put a single module as the project itself. This does not respect the OCA standards. However, your work for migrating this module is very appreciated. I will continue with your branch and finish migrating this.

**Comment by andreschenkels** [comment]
David how far did you get with the work? I would like to help you on this. Porting to 8.0 and 9.0

**Comment by ghost** [comment]
@aschenkels-ictstudio Sorry for replying late. I migrated the module from 7.0 to 8.0 here https://github.com/OCA/connector-redmine/pull/10.

**Pull request closed without merging**